diff --git a/src/modules/engines/software_8/evas_engine.c b/src/modules/engines/software_8/evas_engine.c
index 367dd5c..d2788fb 100644
--- a/src/modules/engines/software_8/evas_engine.c
+++ b/src/modules/engines/software_8/evas_engine.c
@@ -341,10 +341,21 @@ eng_image_cache_get(void *data __UNUSED__)
 static void
 eng_font_draw(void *data __UNUSED__, void *context, void *surface,
       Evas_Font_Set *font, int x, int y, int w __UNUSED__, int h __UNUSED__,
-      int ow __UNUSED__, int oh __UNUSED__, const Evas_Text_Props *text_props)
+      int ow __UNUSED__, int oh __UNUSED__, Evas_Text_Props *text_props)
 {
+   static RGBA_Image    *im = NULL;
+   Soft8_Image         *dst = surface;
+
+   im = (RGBA_Image *) evas_cache_image_empty(evas_common_image_cache_get());
+   evas_cache_image_surface_alloc(&im->cache_entry, dst->cache_entry.w, dst->cache_entry.h);
+   evas_common_draw_context_font_ext_set(context,
+					 surface,
+					 evas_common_soft8_font_glyph_new,
+					 evas_common_soft8_font_glyph_free,
+					 evas_common_soft8_font_glyph_draw);
+
    evas_common_font_draw_prepare(text_props);
-   evas_common_font_draw(surface, context, x, y, text_props);
+   evas_common_font_draw(im, context, x, y, text_props);
    evas_common_draw_context_font_ext_set(context, NULL, NULL, NULL, NULL);
 }
 
