diff --git a/src/modules/engines/software_8/evas_engine.c b/src/modules/engines/software_8/evas_engine.c
index 367dd5c..5bb938b 100644
--- a/src/modules/engines/software_8/evas_engine.c
+++ b/src/modules/engines/software_8/evas_engine.c
@@ -339,16 +339,6 @@ eng_image_cache_get(void *data __UNUSED__)
 }
 
 static void
-eng_font_draw(void *data __UNUSED__, void *context, void *surface,
-      Evas_Font_Set *font, int x, int y, int w __UNUSED__, int h __UNUSED__,
-      int ow __UNUSED__, int oh __UNUSED__, const Evas_Text_Props *text_props)
-{
-   evas_common_font_draw_prepare(text_props);
-   evas_common_font_draw(surface, context, x, y, text_props);
-   evas_common_draw_context_font_ext_set(context, NULL, NULL, NULL, NULL);
-}
-
-static void
 eng_font_cache_flush(void *data __UNUSED__)
 {
    evas_common_font_flush();
@@ -409,7 +399,6 @@ module_open(Evas_Module * em)
    ORD(image_cache_flush);
    ORD(image_cache_set);
    ORD(image_cache_get);
-   ORD(font_draw);
    ORD(image_scale_hint_set);
    ORD(image_scale_hint_get);
 
diff --git a/src/modules/engines/software_8_x11/evas_engine.c b/src/modules/engines/software_8_x11/evas_engine.c
index c8efb24..bb1c9df 100644
--- a/src/modules/engines/software_8_x11/evas_engine.c
+++ b/src/modules/engines/software_8_x11/evas_engine.c
@@ -637,6 +637,28 @@ eng_canvas_alpha_get(void *data __UNUSED__, void *context __UNUSED__)
    return EINA_FALSE;
 }
 
+static void
+eng_font_draw(void *data, void *context, void *surface,
+      Evas_Font_Set *font, int x, int y, int w __UNUSED__, int h __UNUSED__,
+      int ow __UNUSED__, int oh __UNUSED__, Evas_Text_Props *text_props)
+{
+   static RGBA_Image    *im = NULL;
+   Soft8_Image         *dst = surface;
+   Render_Engine *re = (Render_Engine *)data;
+
+   im = (RGBA_Image *) evas_cache_image_empty(evas_common_image_cache_get());
+   im->cache_entry.w = re->w;
+   im->cache_entry.h = re->h;
+   evas_common_draw_context_font_ext_set(context,
+					 surface,
+					 evas_common_soft8_font_glyph_new,
+					 evas_common_soft8_font_glyph_free,
+					 evas_common_soft8_font_glyph_draw);
+   evas_common_font_draw_prepare(text_props);
+   evas_common_font_draw(im, context, x, y, text_props);
+   evas_common_draw_context_font_ext_set(context, NULL, NULL, NULL, NULL);
+}
+
 /* module advertising code */
 static int
 module_open(Evas_Module * em)
@@ -672,6 +694,7 @@ module_open(Evas_Module * em)
    ORD(output_redraws_next_update_push);
    ORD(output_flush);
    ORD(output_idle_flush);
+   ORD(font_draw);
    /* now advertise out own api */
    em->functions = (void *)(&func);
    return 1;
