--- a/src/lib/ecore_evas/ecore_evas_x.c	2013-02-13 17:30:11.517455900 +0100
+++ b/src/lib/ecore_evas/ecore_evas_x.c	2013-02-13 17:27:04.487442612 +0100
@@ -133,6 +133,31 @@ _ecore_evas_x_sync_clear(Ecore_Evas *ee)
    ee->engine.x.sync_counter = 0;
 }
 
+#ifdef HAVE_ECORE_X_XCB
+static xcb_visualtype_t *
+xcb_visualtype_get(xcb_screen_t *screen, xcb_visualid_t visual)
+{
+	xcb_depth_iterator_t  iter_depth;
+
+	if (!screen) return NULL;
+
+	iter_depth = xcb_screen_allowed_depths_iterator(screen);
+	for (; iter_depth.rem; xcb_depth_next (&iter_depth))
+	{
+		xcb_visualtype_iterator_t iter_vis;
+
+		iter_vis = xcb_depth_visuals_iterator(iter_depth.data);
+		for (; iter_vis.rem; --screen, xcb_visualtype_next (&iter_vis))
+		{
+			if (visual == iter_vis.data->visual_id)
+				return iter_vis.data;
+		}
+	}
+
+	return NULL;
+}
+#endif /* HAVE_ECORE_X_XCB */
+
 # ifdef BUILD_ECORE_EVAS_OPENGL_X11
 static Ecore_X_Window
 _ecore_evas_x_gl_window_new(Ecore_Evas *ee, Ecore_X_Window parent, int x, int y, int w, int h, int override, int argb, const int *opt)
